hrabsence/tests/phpunit/CRM/HRAbsence/BAO/HRAbsenceEntitlementTest.php

```diff
@@ -90,4 +90,67 @@ public function testRecalculateAbsenceEntitlementsForPeriodCreatesTheContactsEnt
     $this->assertEquals(23, $contact2Entitlement->amount);
   }
 
+  public function testRecalculateAbsenceEntitlementDoesNotOverwriteExistingEntitlements() {
+    $absenceType = AbsenceTypeFabricator::fabricate();
+
+    $contact = ContactFabricator::fabricate([]);
```

Comment:
you don't need to pass empty array here , contact fabricator already have an empty array as default parameter.